### DISCUSSION TITLE
Gradually low charging current when temperature gets high

### DISCUSSION THREAD

**Discussion by Imaginous** [serkri/SmartEVSE-3#235] [Ideas]
I'm charging at 32A on a single phase. I observe that on long charging runs the system slowly heats up and at the max temperature shutdown.
This is perfectly fine... but...

It would be great if a "max" temperature range could be supplied in which the charging current is linear lowered with the temperature.

Example:
Max temperature range 55-65
Current range 6-32A (min/max system current)

Up till 55 degrees the system tries to charge at 32A (max current)
Between 55-65 it lowers the max current by 32-6=26 devided by the delta temperature 65-55=10, so 2.6A per degree from 55 degrees. So 60 degrees would lead to a current of 32-13A=19A.

If it still would reach 65 it would shutdown.

It's better for thr battery of the car to keep charging instead of cutting of.

**Comment by dingo35** [comment]
You are aware you can change MAX_TEMP on the LCD screen?

**Reply by Imaginous** [reply]
Yes I know, it's set to 65... Don't want to go higher ;)

**Comment by dingo35** [comment]
Second, if your charger is not heated up by the sun but by large currents, you better find the connection or part that heats up and fix the problem!!!!

**Reply by Imaginous** [reply]
The "problem" is that at 32A even 6mm2 cables heat up. Also the contactor relay heats up.

I cannot find any extremely hot parts, but mostly a buildt up of heat.
I will add extra ventilation today, since the box is located in the shed.

But 32A via one wire pair generates far more heat (and loss) then a 3 phase solution.

The idea is that you can prevent a protective shutdown in single phase installations.

**Reply by dingo35** [reply]
You've got to find your heat source; 32A is not extreme, and I have _never_ seen power panels in residential environments needing ventilation.

As a reference: in the US I have a 200A house connection, with 40A and 60A devices semi-permanently running, without _any_ heat development.

It's impossible that 6mm2 wiring running 32A is generating heat; are you using ferules, are you sure your cables are certified, are you sure your contactor is not Chinese crap?

To my knowledge you are the only SmartEVSE user (of thousands) that thinks its normal its device switches off because of heat internally generated in the power panel....

**Reply by dingo35** [reply]
If you are Dutch speaking, post pictures of your installation on https://gathering.tweakers.net/forum/list_message/77794188#77794188; hopefully experts there can give you feedback on your wiring....
